StudentManager/Controllers/DataProvider.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Newtonsoft.Json;
+using StudentManager.Models;
+
+namespace StudentManager.Controllers
+{
+    public class DataProvider
+    {
+        public List<Student> Students { get; set; }
+        public List<Class> Classes { get; set; }
+        public List<Course> Courses { get; set; }
+        public List<Score> Scores { get; set; }
+        public List<Subject> Subjects { get; set; }
+        public List<Teacher> Teachers { get; set; }
+
+        private static DataProvider instance;
+        public static DataProvider GetInstance => instance ?? (instance = new DataProvider());
+
+        private DataProvider()
+        {
+            Students = new List<Student>();
+            Classes = new List<Class>();
+            Courses = new List<Course>();
+            Scores = new List<Score>();
+            Subjects = new List<Subject>();
+            Teachers = new List<Teacher>();
+        }
+
+        public void LoadData()
```

Comment:
The directory is always `Data`, you can declare a constant `DataDirectory = "Data"` and access directly from the functions `ReadJson` `SaveJson`